### DISCUSSION TITLE
Documentation doesn't explicitly state that p-tables were built with template forms in mind

### DISCUSSION THREAD

**Discussion by daerogami** [primefaces/community#1272] [PrimeNG]
I have implemented p-tables with editable rows using reactive forms and it has always been a struggle and I've seen other issues raised about it on github. All of the examples in the documentation seem to use template-driven forms which is fine, but I don't understand why there isn't an explicit statement that pTables were intended to be used with template-driven forms or even a disclaimer that reactive forms can make row-editable p-tables unwieldy. Would it be a fair statement that one should be using template-driven forms as a default approach with editable p-table rows?

**Comment by cetincakiroglu** [comment]
Hi,

Thanks for reporting the issue, could you please share a stackblitz example so we can understand better the issue you're having?

**Reply by daerogami** [reply]
https://stackblitz.com/edit/cbsjr2?file=stackblitz.README.txt
Here is a stack blitz demonstrating some of the awkwardness. I still have a few more examples that I will finish building out such as saving row data in the reactive-forms version and making the expanded rows editable, but there should be enough here to demonstrate what I referred to in my initial post. I have comments on both the template and component class and I will provide a compilation of them here for discussion:

- reactive-table.component.ts
  - line 14: These are identical to the domain models except that we use them to apply typing to the equivalent reactive-forms table component
  - line 58: Form initialization is all additional compared to template driven forms
  - line 75: Notice how every row edit interaction is complicated by both digging through the domain model object _and_ the formGroup hierarchy
  - line 116: Both of these mapping methods are required to load data into our formGroups which aren't needed for template-driven forms
  - line 119: rowId is necessary to uniquely identify rows since new items may not be assigned an id from the server; this is also necessary to ensure row-expand works as well
- reactive-table.component.html
  - line 1: Notice how any and every reference except reactive control name attributes have to use control accessors and then access the value
  - line 44: formControlName is the only thing in templates that remains simple with reactive forms when using editable rows
  - line 67: Items that aren't editable still must be awkwardly stored and loaded from the formGroup
  - line 131: because we must contextually reference the array from within a template, we have to bind the template variable "ri" instead of using "models" as a literal value

**Comment by vanuatoo** [comment]
I'm struggling to work cell edit mode with Reactive Forms. In my case part of the data is coming from the back-end and I just want to have one column editable (cell edit mode). I'm trying to use FormArray. Unfortunately the documentation is only showing Template-Driven Forms.

**Comment by AlejandroFlorin** [comment]
+1 There is no documentation using Reactive form. Very difficult to figure out. I've given up and used a popup editor
